src/Generator.cpp

```diff
@@ -21,362 +26,655 @@
 #include "HGCalSimulation/FastShower/interface/Event.h"
 #endif
 
+using namespace std;
+
+Generator::Generator(const Parameters& params): geometry_(params.geometry()),
+                                                output_(params.general().output_file),
+                                                shower_(params.shower()),
+                                                parameters_(params) { 
+}
+
+
+Generator::~Generator(){
+}
+
+
+double** Generator::readCalibration(const string filename) {
+
+    double** calib = new double*[52];
+    for(int i = 0; i < 52; ++i)
+        calib[i] = new double[3];
+
+    ifstream fichier(filename);
+    string line;
+
+    if (fichier) {
+        while(getline(fichier, line)) {
+
+            // Remove comment lines (beginning with #)
+            if (line.find("#") != std::string::npos) {
+                continue;
+            }
+            else {
+                int layer_col;
+                double mip[3];
+                double noise[3];
+
+                istringstream strm(line);
+
+                strm >> layer_col;
+                strm >> mip[2];
+                strm >> mip[1];
+                strm >> mip[0];
+
+                strm >> noise[2];
+                strm >> noise[2];
+                strm >> noise[2];
+                strm >> noise[2];
+                strm >> noise[1];
+                strm >> noise[0];
+
+                for (int i = 0; i < 3; i++) {
```

Comment:
Use a const variable with a clear name instead of magic number `3`

Reply:
Fixed